libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1728,10 +1728,39 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	else
 		_operation.subExpression().accept(*this);
 	Type const* subExprType = type(_operation.subExpression());
-	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
-	if (!result)
+
+
+	// Check if the operator is built-in or user-defined.
+	Result<FunctionDefinition const*> userDefinedOperatorResult = subExprType->userDefinedOperator(
```

Comment:
This actually returns the underlying function, not the operator, so how about renaming it like this?
```suggestion
	Result<FunctionDefinition const*> userDefinedOperatorResult = subExprType->operatorDefinition(
```

Reply:
done